# Amplicon sequencing analysis for Yeast

We used the Natrix workflow to process the amplicon sequencing data. This performs fastqc to quality control the reads and DADA2 to generate a count table. I then vst transformed the count table using DESeq2 and performed hierarchical clustering and PCA to detect any outliers. Since no outliers were detected, I added all the counts from the technical replicates together per sample in the raw count table, and RPM normalized this table. Then I subtracted the RPM counts of the control starting library to allow for a better cross sample comparison. This is the data frame that is used as input in the data_analysis notebook. Both nucleotide and amino acid versions of the tables and results have been generated.

## Directory structure
```
.
├── Natrix  # Natrix workflow
│
├── data
│   ├── interim                                     # contains the output of the Natrix pipeline.
│   │   ├── assembly
│   │   ├── finalData                               # contains the raw count table output from DADA2
│   │   ├── logs
│   │   └── qc                                      # contains the fastqc output
│   ├── processed                                   # Normalized count tables and analysis outputs.
│   │   ├── aa_rpm_minus_control_expression.csv
│   │   ├── aa_rpm_normalized_reads.csv
│   │   ├── aa_upset_data.csv
│   │   ├── deseq2
│   │   ├── nt_rpm_minus_control_expression.csv
│   │   ├── nt_rpm_normalized_counts.csv
│   │   ├── nt_upset_data.csv
│   │   ├── vst_transformed_combined_reads.csv
│   │   └── vst_transformed_reads.csv
│   ├── raw                                         # Raw data as input for Natrix. (Too big to upload)
│   │   ├── 1Code
│   │   ├── reads
│   │   ├── sample_sheet.csv
│   │   └── sample_sheet_no_controls.csv
│   └── test
│       ├── 1FgAgo1_S1_L001_A_R1_001.fastq.gz
│       ├── 1FgAgo1_S1_L002_A_R1_001.fastq.gz
│       ├── 1FgAgo1_S1_L003_A_R1_001.fastq.gz
│       └── 1FgAgo1_S1_L004_A_R1_001.fastq.gz
├── environment.yml                                 # conda environment file to work with the python notebooks. Note this environment is actually much larger than what is needed, so feel free to make your own.
├── environment_R.yml                               # conda environment file to work with the R scripts.
├── notebooks                                       # contains the jupyter notebooks.
│   ├── 1Code
│   │   ├── DataAnalysis.py
│   │   └── alignment.py
│   ├── DESeq2.R                                    # R script to perform DESeq2 analysis.
│   ├── count_table.ipynb                           # notebook to generate the count table, slightly modified from Natrix workflow to work with the large dataset.
│   ├── data_analysis.ipynb                         # notebook to perform the data analysis.
│   ├── data_processing.ipynb                       # notebook to process the data, mainly normalization and checking replication.
│   └── upset_plots.R                               # R script to generate the upset plots.
├── output_tree.txt                                 # full directory structure.
└── reports
    ├── 20240522_pheromone_illumina_library_MJ.pptx
    ├── figures                                     # contains the figures generated by the notebooks.
    │   ├── aa_top_10_sequences_per_species.pdf
    │   ├── aa_top_5_sequences_per_sample.pdf
    │   ├── aa_upset_plot.pdf
    │   ├── differentially_expressed_seqs_hm.pdf
    │   ├── nt_global_clustering.pdf
    │   └── nt_upset_plot.pdf
    └── ~$20240522_pheromone_illumina_library_MJ.pptx

32 directories, 140 files
```
